Fix

spaskob opened the pull request.

Commit:
Merge #44900

44900: storage/cloud: Make it possible to disable external http storage r=miretskiy a=miretskiy

Informs #44320

Make it possible to disable external http storage.

This change enables the operators to disable access
 to external http servers as well as custom http endpoints
 overrides for cloud storage implementations.

Release note (security update): Make it possible for operators
to disable external http access when performing certain
operations (BACKUP, IMPORT, etc).

The external http access, as well as custom http endpoints, are
disabled by providing an --external-io-disable-http flag.

This flag provides a light weight option to disable http external
access in the environments where running a full fledged proxy
server may not be feasible.  If running a proxy service is
acceptible, operators may choose to start cockroach binary
specifying HTTP(s)_PROXY environment setting.

Co-authored-by: Yevgeniy Miretskiy <yevgeniy@cockroachlabs.com>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45316)
<!-- Reviewable:end -->

spaskob committed:
job: subtle bug in CANCEL JOB when the job is reverting

Previously, trying to cancel a job on a node that was not running it
will not work. Also, if reverting job was resumed it was not passed
the failure from Resume.

This PR fixes this by adding a similar if case for checking if
a job is CancelRequested in the adopt loop. This also makes the code
more readable. Also added a specific error to indicate job was cancelled
by user to be able to distinguishing if we are reverting a failed or
cancelled job. Before this we were relying on the absence of error for
that and this was error prone.

Release note (bug fix): in some rare cases a job was not cancellable
when in state Reverting.
